src/System.Private.CoreLib/src/System/MulticastDelegate.cs

```diff
@@ -135,32 +136,35 @@ public override sealed bool Equals(object obj)
         private bool InvocationListEquals(MulticastDelegate d)
         {
             Debug.Assert(d != null && (_invocationList as object[]) != null, "bogus delegate in multicast list comparison");
-            object[] invocationList = _invocationList as object[];
+            object[] invocationList = (_invocationList as object[])!; // Calling code ensures invocationList is object[]
```

Comment:
```suggestion
            object[] invocationList = (object[])_invocationList; // Calling code ensures invocationList is object[]
```

To match other places in this file?